src/Ojs/CoreBundle/Filter/CatalogFilter.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Ojs\CoreBundle\Filter;
+
+use Doctrine\ORM\Mapping\ClassMetaData;
+use Doctrine\ORM\Query\Filter\SQLFilter;
+
+class CatalogFilter extends SQLFilter
+{
+    private $catalogs = [];
+
+    public function setCatalogs($catalogs = [])
+    {
+        $this->catalogs = $catalogs;
+    }
+
+    public function addFilterConstraint(ClassMetadata $targetEntity, $targetTableAlias)
+    {
+        //if params have asterisk sign, pass catalog filter
+        if(in_array('*', $this->catalogs)){
+            return '';
+        }
+
+        $mappings = $targetEntity->getAssociationMappings();
+        if(!array_key_exists('catalogs', $mappings) ||
```

Comment:
For my opinion, checking eligibility of entity via explicit inheritance is a better choice.

Like `$targetEntity->reflClass->implementsInterface('CatalogAware')`